File: app/src/main/java/com/maple/audiometry/utils/ArrayUtils.java
package com.maple.audiometry.utils;

import java.util.List;

/**
 * 数组工具类
 *
 * @author shao
 */
public class ArrayUtils {

    /**
     * 求一个数组的总和
     */
    public static int sum(int[] datas) {
        int sum = 0;
        for (int s : datas) {
            sum += s;
        }
        return sum;
    }

    /**
     * 求一个数组的总和
     */
    public static double sum(List<Double> datas) {
        double sum = 0;
        for (int i = 0; i < datas.size(); i++) {
            sum += datas.get(i);
        }
        return sum;
    }

    /**
     * 求一个数组的平均数
     */
    public static int avg(int[] datas) {
        if (datas.length == 0) {
            return 0;
        } else {
            return sum(datas) / datas.length;
        }
    }

    /**
     * 求一个数组的平均数
     */
    public static double avg(List<Double> datas) {
        if (datas.size() == 0) {
            return 0;
        } else {
            return sum(datas) / datas.size();
        }
    }

    /**
     * 截取数组
     */
    public static double[] sub(List<Double> allVolume, int num) {
        double[] dous = new double[num];
        if (allVolume.size() < num) {// 长度不够
            for (int i = 0; i < allVolume.size(); i++) {
                dous[i] = allVolume.get(i);
            }
        } else {// 长度足够。选取后半部分
            int index = allVolume.size() - num;
            for (int i = 0; i < num; i++) {
                dous[i] = allVolume.get(index + i);
            }
        }
        return dous;
    }
}

File: app/src/main/java/com/maple/audiometry/utils/ArrayUtils.kt
package com.maple.audiometry.utils

/**
 * 数组工具类
 *
 * @author shao
 */
object ArrayUtils {

    /**
     * 求一个数组的总和
     */
    @JvmStatic
    fun sum(datas: IntArray): Int {
        var sum = 0
        for (s in datas) {
            sum += s
        }
        return sum
    }

    /**
     * 求一个数组的总和
     */
    @JvmStatic
    fun sum(datas: List<Double>): Double {
        var sum = 0.0
        for (i in datas.indices) {
            sum += datas[i]
        }
        return sum
    }

    /**
     * 求一个数组的平均数
     */
    @JvmStatic
    fun avg(datas: IntArray): Int {
        return if (datas.isEmpty()) {
            0
        } else {
            sum(datas) / datas.size
        }
    }

    /**
     * 求一个数组的平均数
     */
    @JvmStatic
    fun avg(datas: List<Double>): Double {
        return if (datas.isEmpty()) {
            0.0
        } else {
            sum(datas) / datas.size
        }
    }

    /**
     * 截取数组
     */
    @JvmStatic
    fun sub(allVolume: List<Double>, num: Int): DoubleArray {
        val dous = DoubleArray(num)
        if (allVolume.size < num) {// 长度不够
            for (i in allVolume.indices) {
                dous[i] = allVolume[i]
            }
        } else {// 长度足够。选取后半部分
            val index = allVolume.size - num
            for (i in 0 until num) {
                dous[i] = allVolume[index + i]
            }
        }
        return dous
    }
}
